### PR TITLE
fix: frontend audit — double-send guard, SSE timeout, cleanup

### PR DESCRIPTION
**Pull request by jdguggs10**
## Summary
- **Double-send guard**: Disable send button and block Enter key during active stream; whitespace-only input also blocked. Belt-and-suspenders guard in `handleSendMessage`.
- **SSE idle timeout**: 60s resettable timeout on `reader.read()` via `Promise.race`. On timeout, cancels the reader and shows a user-facing recovery message ("Connection timed out. Please try again."). Timeout cleanup in `finally` block.
- **Setter rename**: `setinputMessageText` → `setInputMessageText` (proper camelCase, 4 occurrences).
- **Remove `katex`**: Unused dependency removed from `web/package.json`.
- **ESLint `no-explicit-any`**: Changed from `"off"` to `"warn"` — surfaces `any` usage during development (46 warnings, 0 errors).
- **`cache: 'no-store'`**: Added to ESPN/Yahoo/Sleeper league fetches and preferences GET to prevent edge-case caching.

## Test plan
- [ ] Open `/chat`, send a message, verify streaming works end-to-end
- [ ] Press Enter rapidly during a response — confirm no double-send (button disabled, Enter blocked)
- [ ] Verify tool calls still render correctly
- [ ] `npx tsc --noEmit` passes
- [ ] `npm run lint` passes (0 errors, 46 `any` warnings expected)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: web/lib/chat/assistant.ts [hunk 1]
```diff
@@ -140,13 +142,24 @@ export const handleTurn = async (
...
     let buffer = "";
 
+    const READ_TIMEOUT_MS = 60_000;
+    const readWithTimeout = (): Promise<ReadableStreamReadResult<Uint8Array>> => {
+      clearTimeout(timeoutId);
+      return Promise.race([
+        reader!.read(),
+        new Promise<never>((_, reject) => {
+          timeoutId = setTimeout(() => reject(new Error('SSE_READ_TIMEOUT')), READ_TIMEOUT_MS);
+        }),
```
[the remark this answers is not in the document]
Comment (jdguggs10) [reply web/lib/chat/assistant.ts:167]: Addressed in 8fd7526a: increased idle timeout to 300_000ms to reduce false positives on long-running tool calls while keeping stall protection. The timeout safeguard remains in place, but is now much less likely to interrupt valid quiet periods.

File: web/lib/chat/assistant.ts [hunk 2]
```diff
@@ -982,6 +1000,20 @@ export const processMessages = async (controller?: AbortController) => {
...
+    if (error instanceof Error && error.message === 'SSE_READ_TIMEOUT') {
+      mutateChatMessages((chatMessages) => {
+        chatMessages.push({
+          type: "message",
+          role: "assistant",
+          content: [{ type: "output_text", text: "Connection timed out. Please try again." }],
+        });
+      });
+      setLoadingState({ status: "idle", thinkingText: "" });
+      return;
```
[the remark this answers is not in the document]
Comment (jdguggs10) [reply web/lib/chat/assistant.ts:1028]: Addressed in 8fd7526a: timeout recovery now clears queued conversation items (`setConversationItems([])`) before returning to idle, preventing stale timed-out turns from being replayed on retry in stored-responses flow.

**Comment by jdguggs10** [comment]
Addressed follow-up review items in `5e39659b`:

- Suppressed expected timeout noise in `handleTurn` (`SseTimeoutError` no longer logs as `console.error`).
- Added idle-state reset in approval-flow error path (`handleApprovalResponse`) to avoid stuck non-idle UI after non-timeout failures.

Previously addressed in `8fd7526a`:
- Increased SSE idle timeout to 300s to reduce false positives for long-running tool calls.
- Cleared queued `conversationItems` on timeout to prevent replay of stale timed-out turns.

Lockfile churn note: changes came from `npm uninstall katex -w web` with the current npm version in this workspace and are intentional for this dependency removal.
